### PR TITLE
Build on gha

### PR DESCRIPTION
**Pull request by paoliniluis**
This PR tends to fix these 3 issues:
- https://github.com/metabase/metabase-qa/issues/28
- https://github.com/metabase/metabase-qa/issues/29
- https://github.com/metabase/metabase-qa/issues/30

### CODE REVIEW

File: dbs/postgres/latest/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM postgres:11-alpine
+FROM postgres:latest
```
Comment (metamben) [inline dbs/postgres/latest/Dockerfile:1]: Why are we abandoning apline?

**Review by metamben — COMMENTED** [review]
- what's going to happen to the removed mongo versions?
- mogo's `build.sh` is a bit more complicated than how we build with GH, is that not a problem?
- the `build.sh` scripts are not removed, what's the plan for them?
- the oracle Dockerfile need changes in the paths, I guess
- this PR changes the name of the oracle 21 image, this will have to be reflected in the testing workflow, right?

File: dbs/postgres/latest/Dockerfile (hunk 1 above)
Comment (paoliniluis) [reply dbs/postgres/latest/Dockerfile:1]: postgres doesn't have a latest-alpine tag unfortunately, so I'm relying on the standard image

**Comment by paoliniluis** [comment]
> * what's going to happen to the removed mongo versions?
nothing, as we'll keep the images in Dockerhub. We just remove the recipes, the cakes are already in the fridge

> * mongo's `build.sh` is a bit more complicated than how we build with GH, is that not a problem?
I saw that we build intermediate images and then do a "commit" and I couldn't understand why as if it was for speed, the images built via the Dockerfile are quick enough

> * the `build.sh` scripts are not removed, what's the plan for them?
We can remove those if needed, I would like to leave only Dockerfiles

> * the oracle Dockerfile need changes in the paths, I guess
whoops, you're right

> * this PR changes the name of the oracle 21 image, this will have to be reflected in the testing workflow, right?
Yes, you're right

**Comment by metamben** [comment]
> > * what's going to happen to the removed mongo versions?
> 
> nothing, as we'll keep the images in Dockerhub. We just remove the recipes, the cakes are already in the fridge

What's the reason for keeping the mongo 4.2 version then? (In fact, I prefer keeping the recipes too.)

**Comment by paoliniluis** [comment]
> > > * what's going to happen to the removed mongo versions?
> > 
> > 
> > nothing, as we'll keep the images in Dockerhub. We just remove the recipes, the cakes are already in the fridge
> 
> What's the reason for keeping the mongo 4.2 version then? (In fact, I prefer keeping the recipes too.)

that's the minimum version we support right now right?

regarding the recipes: ok, doesn't hurt my feelings

**Comment by paoliniluis** [comment]
ok, so I did a few changes:
1) now the Dockerfile is just 1, no more Dockerfile per folders, we pass the tag dynamically on the build
2) fixed the Oracle paths @metamben

File: dbs/oracle/Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 FROM gvenzl/oracle-xe:21.3.0-slim
 
-COPY --chown=oracle:dba listener.ora sqlnet.ora /opt/oracle/homes/OraDBHome21cXE/network/admin/
-COPY --chown=oracle:dba server /opt/oracle/server
-COPY --chown=oracle:dba user.sql /container-entrypoint-initdb.d/
+COPY --chown=oracle:dba dbs/oracle/listener.ora sqlnet.ora /opt/oracle/homes/OraDBHome21cXE/network/admin/
```
Comment (metamben) [inline dbs/oracle/Dockerfile:3]:
```suggestion
COPY --chown=oracle:dba dbs/oracle/listener.ora dbs/oracle/sqlnet.ora /opt/oracle/homes/OraDBHome21cXE/network/admin/
```

**Review by metamben — COMMENTED** [review]
Looks good! Now if we added a matrix listing the different versions we build, we would have the recipes of all images in use in the GH workflows and the `build.sh` could be deleted.

**Review by metamben — APPROVED** [review]
This is already a big improvement! 👍

**Comment by paoliniluis** [comment]
This one should be good to go

**Pull request merged**